tests/trainer/test_trainer.py

```diff
@@ -3451,3 +3456,103 @@ def test_hyperparameter_search_backends(self):
             list(ALL_HYPERPARAMETER_SEARCH_BACKENDS.keys()),
             list(HPSearchBackend),
         )
+
+
+@require_peft
+@require_torch
+class TrainerPeftTest(unittest.TestCase, TrainerIntegrationCommon):
+    def get_regression_adapter_trainer(
+        self,
+        keep_report_to=False,
+        adapter_name="default",
+        adapter_args=None,
+        **kwargs,
+    ):
+        # from transformers import OPTForCausalLM
+        base_model_id = "hf-internal-testing/tiny-random-OPTForCausalLM"
+        pretrained_base_model = AutoModelForCausalLM.from_pretrained(base_model_id).to(torch_device)
+        tokenizer = AutoTokenizer.from_pretrained(base_model_id)
+
+        def m(sample):
+            return tokenizer(
+                sample["quote"],
+                return_tensors="pt",
+                padding="max_length",
+                truncation=True,
+                max_length=tokenizer.model_max_length,
+            )
+
+        from datasets import load_dataset
+
+        data = load_dataset("Abirate/english_quotes")
+        data = data["train"].filter(lambda example, indice: indice < 9, with_indices=True).map(m, batched=True)
+
+        # generate peft
+        from peft import LoraConfig, get_peft_model
+
+        adapter_args = adapter_args or {
+            "r": 8,
+            "lora_alpha": 16,
+            "lora_dropout": 0.05,
+            "bias": "none",
+            "target_modules": ["q_proj"],
+            "task_type": "CAUSAL_LM",
+        }
+
+        lora_config = LoraConfig(**adapter_args)
+
+        peft_model = get_peft_model(
+            model=pretrained_base_model,
+            peft_config=lora_config,
+            adapter_name=adapter_name,
+        )
+
+        compute_metrics = kwargs.pop("compute_metrics", None)
+        optimizers = kwargs.pop("optimizers", (None, None))
+        output_dir = kwargs.pop("output_dir", "./regression")
+        preprocess_logits_for_metrics = kwargs.pop("preprocess_logits_for_metrics", None)
```

Comment:
It's best to keep logic for tests as simple and explicit as possible. Here, rather than popping from kwargs, these should be kwargs with default values in `get_regression_adapter_trainer`

Reply:
Basically you are right. I imitated the coding style of other test case in test_trainer.py, there mixes some arges in kwargs and pops them later. What do you think? To keep the coding style identity, or I could seperate them in method args, that's not a big effort.

Reply:
I'd rather they were explicitly set, or, tbh because this method is only used in one place, just hardcoded